server/notebooks/api_views.py

```diff
@@ -80,9 +81,29 @@ def perform_destroy(self, instance):
 
     def perform_create(self, serializer):
         ctx = self.get_serializer_context()
-        notebook_owner_id = Notebook.objects.values_list("owner", flat=True).get(
-            id=ctx["notebook_id"]
-        )
-        if self.request.user.id != notebook_owner_id:
+        notebook = Notebook.objects.get(id=ctx["notebook_id"])
+
+        # check if user has permission to create a revision
+        if self.request.user.id != notebook.owner_id:
             raise PermissionDenied
```

Comment:
This seems like something that would be used in a lot of places perhaps? If so, this could be pulled out into a view decorator. Similar to Django's `@staff_member_required` but `@notebook_owner_required` or something similar.

Reply:
Looking closer this isn't a simple Django view, it's part of the `ModelViewSet`, but maybe DRF has hooks to define something similar as a permission class that can be re-used?

Reply:
Hmm, I'm not sure how to do this, given that it requires the lookup of a relate piece of metadata? I agree this would be nice to define in something like a decorator, were that possible...

Reply:
In a typical Django view, the decorator would get the same `*args` and `**kwargs` as the view so it would also have to do the `Notebook` query to get the `owner_id`. But as a DRF `ModelViewSet` I'm not sure either.